Fix-User-Guide.md

```diff
@@ -0,0 +1,252 @@
+![logo](https://github.com/culturegraph/metafacture-core/wiki/img/metafacture_small.png)
+
+# Fix User Guide
+
+This document provides an introduction to the Metafacture Fix language (short: Metafix or Fix). Fix is a declarative flow oriented language in which transformations of arbitrary metadata/semi-structured data can be defined using the FIX language. The  Fix language for Metafacture is introduced as an alternative to configuring data transformations with Metamorph. Inspired by Catmandu Fix, Metafix processes metadata not as a continuous data stream but as discrete records. The basic idea is to rebuild constructs from the (Catmandu) Fix language like functions, selectors and binds in Java and combine with additional functionalities from the Metamorph toolbox.
+
+## Part of a metafacture worflow
+Metafacture Fix is a transformation module that can be used in a workflow, for this you have to use this in your pipeline:
+
+Flux-Example:
+```PERL
+   infile
+   | open-file
+   | as-lines
+   | decode-marc21
+   | fix(FLUX_DIR + "fixFile.fix")
+   | encode-json
+   | print
+   ;
+```
+
+- when using the FLUX: 
+- - address the `fix`-module
+- - you can add variables
+- - there are some optiones available
+- - The fix-Transformation can be part of the FLUX `|fix("retain(`245??`)")` - usually useful for short fixes
+- - or it can be separated in a external file, that is called in the FLUX-Process as in the code-snipped above
+- when using it in a JAVA-Process, just add the library to your process
+
+## Record-based and metadata manipulating approach
+While Metafature processes the data as a stream, the fix module does not it buffers the incoming stream to distinct records.
+So that you can manipulate all metadata-elements of a record at once and without focussing about the order of the incoming stream - which was a really big hassle in the stream-based MORPH.
```

Comment:
```suggestion
So that you can manipulate all metadata elements of a record at once and without focussing about the order of the incoming stream - which was a really big hassle in the stream-based MORPH.
```